src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,269 @@
+            disabled: this.disabled,
+            mode: this.props.mode,
+            color: this.state.color,
+            ...(!this.disabled && this.props.mode !== "inline")
+                ? { onClick: this.handleClick }
+                : {}
+        });
+    }
+
+    private handleClick = () => {
+        this.setState({ displayColorPicker: !this.state.displayColorPicker });
+    }
+
+    private handleClose = () => {
+        this.setState({ displayColorPicker: false });
+    }
+
+    private updateColorValue = (color: ColorResult) => {
+        const { format, mxObject, colorAttribute } = this.props;
+        if (color && mxObject && !this.disabled) {
+            if (format === "hex") {
+                mxObject.set(colorAttribute, color.hex);
+            } else if (format === "rgb") {
+                mxObject.set(colorAttribute, `rgb(${color.rgb.r},${color.rgb.g},${color.rgb.b})`);
+            } else {
+                mxObject.set(colorAttribute, `rgba(${color.rgb.r},${color.rgb.g},${color.rgb.b},${color.rgb.a})`);
+            }
+
+            this.setState({ alertMessage: "" });
+        }
+    }
+
+    private getValue = (mxObject?: mendix.lib.MxObject): string => {
+        if (mxObject) {
+            return mxObject.get(this.props.colorAttribute) as string;
+        }
+
+        return "";
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                attr: this.props.colorAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+            this.subscriptionHandles.push(window.mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscriptions = () => {
+        this.setState(this.validateColor(this.props.mxObject));
+    }
+
+    private validateColor = (mxObject?: mendix.lib.MxObject): ColorPickerContainerState => {
+        let message = "";
+        const color = this.getValue(mxObject);
+        if (color && this.props.format === "hex" && !color.includes("#")) {
+            message = `Color value ${color} should be of format '#d0d0d0'`;
+        } else if (color && this.props.format === "rgb" && !color.includes("rgb(")) {
+            message = `Color value ${color} should be of format 'rgb(115,159,159)'`;
+        } else if (color && this.props.format === "rgba" && !color.includes("rgba")) {
+            message = `Color value ${color} should be of format 'rgba(195,226,226,1)'`;
+        }
+
+        return {
+            alertMessage: message,
+            color: (message === "") ? color : "",
```

Comment:
Brackets not necessary... also can be rewriten as `!message ? color : ""`

Reply:
Done